src/main/resources/db/migration/V1_68__create_utilisation_report.sql

```diff
@@ -0,0 +1,307 @@
+ RETURNS TABLE(
+  	juror_number character varying,
+	report_from date,
+	report_to date,
+	service_from date,
+	service_to date,
+	trial character varying,
+	jury_from date,
+	jury_to date,
+	report_date date,
+	within_service boolean,
+	active_trial boolean,
+	attendance boolean,
+	holiday boolean)
+ LANGUAGE plpgsql
+AS $function$
+begin
+
+return query select
+	a.juror_number,
+	p_start_date as report_from,
+	p_end_date as report_to,
+	null::date,
+	null::date,
+	null::character varying,
+	null::date,
+	null::date,
+	a.attendance_date,
+	true,
+	a.sat_on_jury,
+	true,
+	null::boolean
+	from juror_mod.appearance a
+	where a.attendance_date between p_start_date and p_end_date
+	and (a.non_attendance is null or a.non_attendance = false)
+	and (a.no_show is null or a.no_show = false)
+	and a.loc_code= p_loc_code
+	order by a.juror_number, a.attendance_date;
+
+END;
+$function$
+;
+
+
+-- function to return juror participation in a trial within the report window
+CREATE OR REPLACE FUNCTION juror_mod.util_trial_participation_list(p_loc_code text, p_start_date date, p_end_date date)
+ RETURNS TABLE(
+   	juror_number character varying,
+	report_from date,
+	report_to date,
+	service_from date,
+	service_to date,
+	trial character varying,
+	jury_from date,
+	jury_to date,
+	report_date date,
+	within_service boolean,
+	active_trial boolean,
+	attendance boolean,
+	holiday boolean)
+ LANGUAGE plpgsql
+AS $function$
+begin
+
+return query select
+	jt.juror_number,
+	p_start_date as report_from,
+	p_end_date as report_to,
+		null::date,
+	null::date,
+	jt.trial_number,
+	jt.empanelled_date as jury_from,
+	coalesce (jt.return_date, p_end_date) as jury_to,
+	report_dates.report_date,
+	true,
+	case when report_dates.report_date between jt.empanelled_date and coalesce (jt.return_date, p_end_date) then true else false end as active_date,
+	null::boolean,
+	null::boolean
+	from juror_mod.juror_trial jt
+	left join (select * from juror_mod.util_report_report_days_list(p_loc_code, p_start_date, p_end_date)) as report_dates
+	on report_dates.report_date between jt.empanelled_date and coalesce (jt.return_date, p_end_date) -- if no return date then assume still on trial for now
+	where jt.loc_code  = p_loc_code
+	and trim(jt.result) in ('J','R')
+	group by jt.juror_number, jt.trial_number, jt.empanelled_date, jt.return_date, report_dates.report_date
+	order by jt.juror_number, report_dates.report_date;
+
+	END;
+$function$
+;
+
+
+-- utility function to return list of all days within the report window and holiday flag
+CREATE OR REPLACE FUNCTION juror_mod.util_report_report_days_list(p_loc_code text, p_start_date date, p_end_date date)
+ RETURNS TABLE(report_date date, holiday boolean)
+ LANGUAGE plpgsql
+AS $function$
+begin
+
+return query select date(report_day.report_day) as report_date, case when h.holiday is not null then true else false end as holiday
+	from generate_series(p_start_date::date, p_end_date::date, '1 day'::interval) report_day
```

Comment:
Can this be extracted to a cte so we preprocess the dates and only cast them once, e.g.

```
...
return query 

with report_days as (
select generate_series(p_start_date::date, p_end_date::date, '1 day'::interval)::date as report_day)
  
select report_days.report_day as report_date, 
case when h.holiday is not null then true else false end as holiday
from report_days
	left join juror_mod.holiday h
	on report_days.report_day = h.holiday
	where h.loc_code is null or h.loc_code = p_loc_code
	order by report_days.report_day;
```